### DISCUSSION TITLE
Provide an error message if a mock will be ignored because it was imported in a setup file

### DISCUSSION THREAD

**Discussion by richardboehme** [vitest-dev/vitest#4759] [Ideas]
Referring to #1450, we just "wasted" ~1.5 hours finding out that our test was affected by this. This is due to mocks just being ignored without any notice. Without any knowledge about the architecture of Vitest: Would it be possible to provide the user 
a meaningful error message if a mock of an already cached module is created? I suspect it may be affected by the performance penalty mentioned in the issue, but I still wanted to ask if it may be feasible. 

If it would be possible and the maintainers agree on it being valuable, we would be happy to contribute or help to contribute.

**Comment by sheremet-va** [comment]
It should actually be possible to check because the path to the module is resolved in place. We can just check if it's already inside the module cache, not in mock cache.

But I am not sure if there are use cases when this is expected behavior.

**Reply by richardboehme** [reply]
This sounds great! I'm not sure either if there would be a use case where this behavior is expected. I feel like if a developer calls `vi.mock` and the mock just won't do anything this can't be expected behavior. However, I'm not experienced enough using vitest to judge this.
